// @fluentui/react-button/lib-commonjs/components/MenuButton/useMenuButtonStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useMenuButtonStyles_unstable = exports.menuButtonClassNames = void 0;

const react_1 = /*#__PURE__*/require("@griffel/react");

const react_theme_1 = /*#__PURE__*/require("@fluentui/react-theme");

const useButtonStyles_1 = /*#__PURE__*/require("../Button/useButtonStyles");

exports.menuButtonClassNames = {
  root: 'fui-MenuButton',
  icon: 'fui-MenuButton__icon',
  menuIcon: 'fui-MenuButton__menuIcon'
};

const useMenuIconStyles = /*#__PURE__*/react_1.__styles({
  "base": {
    "Bg96gwp": "fez10in"
  },
  "small": {
    "Be2twd7": "f1ugzwwg",
    "Bqenvij": "fvblgha",
    "Bg96gwp": "fwrc4pm",
    "a9b677": "frx94fk"
  },
  "medium": {
    "Be2twd7": "f1ugzwwg",
    "Bqenvij": "fvblgha",
    "Bg96gwp": "fwrc4pm",
    "a9b677": "frx94fk"
  },
  "large": {
    "Be2twd7": "f4ybsrx",
    "Bqenvij": "fd461yt",
    "Bg96gwp": "faaz57k",
    "a9b677": "fjw5fx7"
  },
  "notIconOnly": {
    "Frg6f3": ["fbyavb5", "fm0x6gh"]
  }
}, {
  "d": [".fez10in{line-height:0;}", ".f1ugzwwg{font-size:12px;}", ".fvblgha{height:12px;}", ".fwrc4pm{line-height:var(--lineHeightBase200);}", ".frx94fk{width:12px;}", ".f4ybsrx{font-size:16px;}", ".fd461yt{height:16px;}", ".faaz57k{line-height:var(--lineHeightBase400);}", ".fjw5fx7{width:16px;}", ".fbyavb5{margin-left:var(--spacingHorizontalXS);}", ".fm0x6gh{margin-right:var(--spacingHorizontalXS);}"]
});

const useMenuButtonStyles_unstable = state => {
  const menuIconStyles = useMenuIconStyles();
  state.root.className = react_1.mergeClasses(exports.menuButtonClassNames.root, state.root.className);

  if (state.icon) {
    state.icon.className = react_1.mergeClasses(exports.menuButtonClassNames.icon, state.icon.className);
  }

  if (state.menuIcon) {
    state.menuIcon.className = react_1.mergeClasses(exports.menuButtonClassNames.menuIcon, menuIconStyles.base, menuIconStyles[state.size], !state.iconOnly && menuIconStyles.notIconOnly, state.menuIcon.className);
  }

  useButtonStyles_1.useButtonStyles_unstable({ ...state,
    iconPosition: 'before'
  });
  return state;
};

exports.useMenuButtonStyles_unstable = useMenuButtonStyles_unstable;
