// @fluentui/react-theme/lib-commonjs/global/spacings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.verticalSpacings = exports.horizontalSpacings = exports.spacings = void 0;
exports.spacings = {
  none: '0',
  xxs: '2px',
  xs: '4px',
  sNudge: '6px',
  s: '8px',
  mNudge: '10px',
  m: '12px',
  l: '16px',
  xl: '20px',
  xxl: '24px',
  xxxl: '32px'
};
exports.horizontalSpacings = {
  spacingHorizontalNone: exports.spacings.none,
  spacingHorizontalXXS: exports.spacings.xxs,
  spacingHorizontalXS: exports.spacings.xs,
  spacingHorizontalSNudge: exports.spacings.sNudge,
  spacingHorizontalS: exports.spacings.s,
  spacingHorizontalMNudge: exports.spacings.mNudge,
  spacingHorizontalM: exports.spacings.m,
  spacingHorizontalL: exports.spacings.l,
  spacingHorizontalXL: exports.spacings.xl,
  spacingHorizontalXXL: exports.spacings.xxl,
  spacingHorizontalXXXL: exports.spacings.xxxl
};
exports.verticalSpacings = {
  spacingVerticalNone: exports.spacings.none,
  spacingVerticalXXS: exports.spacings.xxs,
  spacingVerticalXS: exports.spacings.xs,
  spacingVerticalSNudge: exports.spacings.sNudge,
  spacingVerticalS: exports.spacings.s,
  spacingVerticalMNudge: exports.spacings.mNudge,
  spacingVerticalM: exports.spacings.m,
  spacingVerticalL: exports.spacings.l,
  spacingVerticalXL: exports.spacings.xl,
  spacingVerticalXXL: exports.spacings.xxl,
  spacingVerticalXXXL: exports.spacings.xxxl
};
